core/beacon/types.go

```diff
@@ -141,8 +156,10 @@ func decodeTransactions(enc [][]byte) ([]*types.Transaction, error) {
 //	uncleHash = emptyUncleHash
 //	difficulty = 0
 //
-// and that the blockhash of the constructed block matches the parameters.
-func ExecutableDataToBlock(params ExecutableDataV1) (*types.Block, error) {
+// and that the blockhash of the constructed block matches the parameters. Nil
+// Withdrawals value will propagate through the returned block. Empty
+// Withdrawals value must be passed via non-nil, length 0 value in params.
```

Comment:
What is the difference between nil and empty? 
If post-shanghai, doesn't all headers contain (potentially empty) withdrawals? 
I don't see why you need to distinguish between nil and empty.

Reply:
Because this method doesn't have fork information, we use `nil` to denote whether the withdrawals root should be calculated and included in the block.